okta/data_source_okta_group.go

```diff
@@ -18,6 +18,12 @@ func dataSourceGroup() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 			},
+			"type": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     false,
+				Description: "Type of the group. When specified in the terraform resource, will act as a filter when searching for the group",
+			},
```

Comment:
Could you please add a validation function for group types?
https://developer.okta.com/docs/reference/api/groups/#group-type